File: android/core/src/main/kotlin/com/allthingsclaude/battery/core/UsageLevel.kt
package com.allthingsclaude.battery.core

/**
 * Maps a utilization percentage (0–100) to a colour-coded severity level.
 *
 * A direct port of `ios/BatteryKit/UsageLevel.swift`. Thresholds match the macOS
 * app exactly (50 / 75 / 90) and, like the desktop default theme, the ramp stays
 * **within the terracotta family** — escalating by darkening (brand → brandDark →
 * brandDeep) rather than turning red.
 *
 * Colours are plain ARGB [Int]s rather than a Compose `Color` on purpose: this
 * module is Android-free, and an `Int` is what both consumers actually want —
 * `NotificationCompat.ProgressStyle.Segment.setColor` takes one directly, and
 * Compose wraps it with `Color(argb)`.
 */
enum class UsageLevel(val label: String, val color: Int, val deeperColor: Int) {
    /** 0–50% */
    LOW("Good", BatteryPalette.BRAND, BatteryPalette.BRAND_DARK),

    /** 50–75% */
    MODERATE("Moderate", BatteryPalette.BRAND, BatteryPalette.BRAND_DARK),

    /** 75–90% */
    HIGH("High", BatteryPalette.BRAND_DARK, BatteryPalette.BRAND_DEEP),

    /** 90%+ */
    CRITICAL("Critical", BatteryPalette.BRAND_DEEP, BatteryPalette.BRAND_DEEP);

    /** True once usage warrants surfacing an alert / bringing the card forward. */
    val isAlarming: Boolean get() = this == HIGH || this == CRITICAL

    companion object {
        fun from(utilization: Double): UsageLevel = when {
            utilization < 50 -> LOW
            utilization < 75 -> MODERATE
            utilization < 90 -> HIGH
            else -> CRITICAL
        }
    }
}

/**
 * The terracotta palette, copied verbatim from the desktop and iOS apps
 * (`ios/BatteryKit/BatteryColors.swift`). Severity escalates by darkening within
 * the brand — multi-colour is the opt-in "classic" theme on desktop and we ship
 * the branded default.
 */
object BatteryPalette {
    const val BRAND = 0xFFD97757.toInt()
    const val BRAND_DARK = 0xFFB85A3A.toInt()
    const val BRAND_DEEP = 0xFF9A4A2C.toInt()

    /**
     * The system-grey equivalent of SwiftUI's `.secondary`, for the one place
     * the forecast deliberately leaves the brand: no open window at all.
     */
    const val SECONDARY = 0xFF8A8A8E.toInt()

    /** Desktop's `#FAF8F4` / `#191814` surfaces. */
    const val SURFACE_LIGHT = 0xFFFAF8F4.toInt()
    const val SURFACE_DARK = 0xFF191814.toInt()
}

File: ios/BatteryKit/UsageLevel.swift
import SwiftUI

/// Maps a utilization percentage (0–100) to a color-coded severity level.
///
/// Thresholds match the macOS app exactly (50 / 75 / 90) and, like the desktop
/// default theme, the ramp stays **within the terracotta family** — escalating
/// by darkening (`brand` → `brandDark` → `brandDeep`) rather than turning red.
enum UsageLevel: String, CaseIterable, Codable {
    case low       // 0–50%
    case moderate  // 50–75%
    case high      // 75–90%
    case critical  // 90%+

    static func from(utilization: Double) -> UsageLevel {
        switch utilization {
        case ..<50:   return .low
        case 50..<75: return .moderate
        case 75..<90: return .high
        default:      return .critical
        }
    }

    /// Solid fill color for gauges, numerals, and bars.
    var color: Color {
        switch self {
        case .low:      return BatteryPalette.brand
        case .moderate: return BatteryPalette.brand
        case .high:     return BatteryPalette.brandDark
        case .critical: return BatteryPalette.brandDeep
        }
    }

    /// A two-stop gradient (color → one shade deeper) for premium ring strokes.
    var gradient: LinearGradient {
        let deep: Color
        switch self {
        case .low, .moderate: deep = BatteryPalette.brandDark
        case .high:           deep = BatteryPalette.brandDeep
        case .critical:       deep = BatteryPalette.brandDeep
        }
        return LinearGradient(colors: [color, deep], startPoint: .topLeading, endPoint: .bottomTrailing)
    }

    /// Accent used on the monochrome Lock Screen accessory slots.
    var accentColor: Color { color }

    /// SF Symbol used in compact surfaces (Dynamic Island, inline widgets).
    var sfSymbol: String {
        switch self {
        case .low:      return "battery.100percent"
        case .moderate: return "battery.75percent"
        case .high:     return "battery.25percent"
        case .critical: return "battery.0percent"
        }
    }

    var label: String {
        switch self {
        case .low:      return "Good"
        case .moderate: return "Moderate"
        case .high:     return "High"
        case .critical: return "Critical"
        }
    }

    /// True once usage warrants surfacing an alert / bringing the activity forward.
    var isAlarming: Bool { self == .high || self == .critical }
}
